lib/presentation/edit_workout/edit_workout_bloc.dart

```diff
@@ -0,0 +1,210 @@
+import 'dart:core';
+
+import 'package:bloc/bloc.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:meta/meta.dart';
+import 'package:stronk/api/workout_repo.dart';
+import 'package:stronk/domain/constants.dart' as Constants;
+import 'package:stronk/domain/model/workout.dart';
+
+import 'ParamContainer.dart';
+
+class EditWorkoutState {
+  final Program programRef;
+  final List<Workout> workoutRef;
+
+  EditWorkoutState({@required this.programRef, this.workoutRef});
+
+  @override
+  String toString() {
+    return ("\n workoutRef: $programRef\n");
+  }
+}
+
+@sealed
+abstract class _Event {}
+
+class InitEvent implements _Event {}
+
+class EditWorkoutEvent implements _Event {
+  String workoutId;
+  String editAction;
+  String newValue;
+
+  EditWorkoutEvent(this.workoutId, this.editAction, this.newValue);
+}
+
+class EditWorkoutExerciseSetsRepsEvent implements _Event {
+  ParamContainer paramContainer;
+  int index;
+
+  EditWorkoutExerciseSetsRepsEvent(this.paramContainer, this.index);
+}
+
+class EditProgramNameEvent implements _Event {
+  // Once the schema supports more than one program
+  //String programId;
+  String newName;
+
+  EditProgramNameEvent(this.newName);
+}
+
+class EditWorkoutBloc extends Bloc<_Event, EditWorkoutState> {
+  final WorkoutRepository workoutRepo;
+
+  @override
+  EditWorkoutState get initialState => new EditWorkoutState(programRef: null);
+
+  @override
+  Stream<EditWorkoutState> mapEventToState(_Event event) async* {
+    var newState = state;
+    if (event is InitEvent) {
+      newState = await handleInit().catchError((error) => print(error));
+    } else if (event is EditWorkoutExerciseSetsRepsEvent) {
+      newState = await handleEditWorkoutExerciseRepsAndSets(
+              event.paramContainer, event.index)
+          .catchError((error) => print(error));
+    } else if (event is EditProgramNameEvent) {
+      newState = await handleEditProgramName(event.newName)
+          .catchError((error) => print(error));
+    } else if (event is EditWorkoutEvent) {
+      newState = await handleEditWorkout(
+          event.workoutId, event.editAction, event.newValue);
+    }
+
+    yield newState;
+  }
+
+  EditWorkoutBloc({@required this.workoutRepo}) {
+    add(new InitEvent());
+  }
+
+  Future<EditWorkoutState> handleInit() async {
+    final activeWorkout = await workoutRepo.retrieveProgram();
+
+    return EditWorkoutState(
+        programRef: activeWorkout, workoutRef: activeWorkout.workouts);
+  }
+
+  Future<EditWorkoutState> handleEditWorkoutExerciseRepsAndSets(
+      ParamContainer paramContainer, int index) async {
+    return _editWorkoutExerciseRepsAndSets(paramContainer, index);
```

Comment:
just curious, but why is the nested call instead of placing the login the this function?